pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/bestpractices/xml/AvoidReassigningLoopVariables.xml

```diff
@@ -0,0 +1,477 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<test-data
+        xmlns="http://pmd.sourceforge.net/rule-tests"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://pmd.sourceforge.net/rule-tests http://pmd.sourceforge.net/rule-tests_1_0_0.xsd">
+    <test-code>
+        <description>violation: reassigned 'for' loop variable</description>
+        <expected-problems>1</expected-problems>
+        <expected-linenumbers>5</expected-linenumbers>
+        <code><![CDATA[
```

Comment:
To verify, that the message is produced correctly and the loop's variable name is used, you can add also an assert for the message:

```xml
<expected-messages>
  <message>Avoid reassigning the loop control variable 'i'</message>
</expected-messages>
```

I think, it's enough, if this is checked for once test case - we don't need to verify this for all.

Reply:
Done!